debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -795,16 +798,48 @@ public OffsetStorageReader offsetStorageReader() {
                                     // this thread is still set as the running thread -> we were not interrupted
                                     // due the stop() call -> probably someone else called the interrupt on us ->
                                     // -> we should raise the interrupt flag
+
                                     Thread.currentThread().interrupt();
                                 }
                                 break;
                             }
-                            catch (RetriableException e) {
-                                LOGGER.info("Retrieable exception thrown, connector will be restarted", e);
+                            catch (RetriableException retriableException) {
+                                LOGGER.info("Retriable exception thrown, connector will be restarted", retriableException);
                                 // Retriable exception should be ignored by the engine
                                 // and no change records delivered.
                                 // The retry is handled in io.debezium.connector.common.BaseSourceTask.poll()
                             }
+                            catch (ConnectException connectException) {
```

Comment:
Shouldn't this rather be limited to `RetriableException`? With this change we'd restart on _all_ exception types, which is not what we want?

Reply:
I thought this was the desired behavior that was discussed on the Jira thread - i.e. the first time the task starts up it will fail fast to provide a clear error. However, if the task was able to connect the first time, then essentially any exception should be handled. The difference being that Retriable Exception will always be retried, while ConnectException is retried based on user settings, defaulting to the current behavior of max-retries=0.
For example, a network issue blocking access to a Postgres source falls under ConnectException today. Otherwise, every connector would need an exhaustive list of exceptions to be defined. This should allow the user to optionally configure Debezium to retry by modifying the settings. It is also backwards compatible with the current behavior for Kafka Connect.
WDYT?

Reply:
@gunnarmorling Let us know if you have concerns here...

Reply:
> I thought this was the desired behavior that was discussed on the Jira thread

Do you have a reference to that? I can't seem to find it (I vaguely remember that we discussed this, but I don't remember this to be the outcome, might be my bad memory though...).

> Otherwise, every connector would need an exhaustive list of exceptions to be defined.

Yes, this has exactly been the strategy so far (see the different connector-specific `ErrorHandler` classes). And I'd say it has served us well so far (indeed the list of retried exceptions grew over time, but it feels rather complete by now).

> For example, a network issue blocking access to a Postgres source falls under ConnectException today. 

What's the exact exception and message? We may add that to the existing PG error handler.

>  This should allow the user to optionally configure Debezium to retry by modifying the settings.

There already is means of doing so, albeit semi-officially only, see [here](https://github.com/debezium/debezium/blob/main/debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java#L480-L489).

All in all, I still believe we should mimick the behavior of Kafka Connect and retry any `RetriableException` based on the user-defined settings you added as part of this PR.

Perhaps we should jump on a call if you still think differently about this? Happy to set something up via Google Meet. Let me know.

Reply:
OK, so we can start with catching RetriableException instead and pinpointing the PG error we encountered when there's no connection to DB.

Reply:
Yes, let's discuss it via Google next week, saw that @zalmane sent an invite, looking forward!